### DISCUSSION TITLE
Self hosted - Setup vortex

### DISCUSSION THREAD

**Discussion by flifloo** [stoatchat/discussions#187] [Q&A] [answered]
Hi,

Recently I am trying to host my own instance of revolt but I have some troubles with the voice server Vortex.

Despite the lack of documentation and docker image, I made my own and set up by digging into the source code, but I encounter some problem and I need some help.

The Docker file:
```Dockerfile
FROM rustlang/rust:nightly-slim AS builder
USER 0:0
WORKDIR /home/rust/src

RUN USER=root cargo new --bin vortex
WORKDIR /home/rust/src/vortex
RUN apt-get update && apt-get install -y build-essential nodejs python libssl-dev pkg-config

COPY Cargo.toml Cargo.lock ./
COPY src ./src
RUN cargo build --release && cargo install --locked --path .

# Bundle Stage
FROM debian:buster-slim
RUN apt-get update && apt-get install -y ca-certificates
COPY --from=builder /usr/local/cargo/bin/vortex ./
EXPOSE 8080
EXPOSE 10000-11000
ENV RTC_IPS 0.0.0.0
ENV RTC_MIN_PORT 10000
ENV RTC_MAX_PORT 11000
ENV HTTP_HOST 0.0.0.0:8080
#ENV WS_URL
#ENV MANAGE_TOKEN
CMD ["./vortex"]
```

I set VOSO_MANAGE_TOKEN, VOSO_PUBLIC_URL and VOSO_WS_HOST for the delta server.
I setup two different proxy location into my nginx for the Vortex HTTP API and the WS:
```nginx
location /vortex/ {
    proxy_pass http://revolt_vortex:8080/;
    proxy_set_header X-Server-IP $server_addr;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Real-HTTPS $https;
    proxy_set_header X-Server-Port $server_port;
    proxy_set_header X-Real-Port $remote_port;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Host $http_host;
}

location /vortex-ws {
    proxy_pass http://revolt_vortex:8080/;
    proxy_set_header X-Server-IP $server_addr;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Real-HTTPS $https;
    proxy_set_header X-Server-Port $server_port;
    proxy_set_header X-Real-Port $remote_port;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Host $http_host;
    proxy_set_header Connection "Upgrade";
    proxy_set_header Upgrade $http_upgrade;
    
    proxy_http_version 1.1;
}
```

On my client side, I success the join_call request, but fail in the web socket communication:
```
VoiceState.ts:125 {error: 1003, message: 'Unable to parse data'}
connect @ VoiceState.ts:125
async function (async)
connect @ VoiceState.ts:99
Fl @ mobx.esm.js:1136
ObservableObject.key @ mobx.esm.js:1120
onClick @ HeaderActions.tsx:114
async function (async)
onClick @ HeaderActions.tsx:112
k @ preact.module.js:1
```

Can someone help me with this ?

**Comment by fatalerrorcoded** [comment, marked as answer]
Vortex was in the middle of being rewritten and had a few bugs that prevented revite from authenticating.

Revite can now connect as of [this commit](https://github.com/revoltchat/vortex/commit/0712a64d52277014942dde3f221b046251569c8a), and vortex now also has a Dockerfile.

**Reply by flifloo** [reply]
Thanks, the server is running without problem and I can connect to a voice channel. I will try later with friends.

**Reply by kiguigui** [reply]
Hey,

can you describe how exactly you got it working? I've tried it and it says the following as Error message.

![grafik](https://user-images.githubusercontent.com/19551960/140073893-66098041-8336-411a-ae3a-85d10a717e51.png)

I'm not even sure what environment variables i need. All other things are working perfectly :)

**Reply by kiguigui** [reply]
I got it working! Like described [here](https://github.com/revoltchat/revolt/discussions/211) you have to set the following environment Variables:


`VOSO_PUBLIC_URL=https://vortex.example.com`
`VOSO_WS_HOST=wss://vortex.example.com`
`VOSO_MANAGE_TOKEN=Mytoken`

`RTC_IPS=0.0.0.0,127.0.0.1`
`RTC_MIN_PORT=13000`
`RTC_MAX_PORT=13010`
`WS_URL=wss://vortex.example.com`


Additionally I've forwarded the Portrange RTC_MIN_PORT to RTC_MAX_PORT to my Docker host, since im exposing these ports on the Docker host.

Edit: Codeblocks
